server/src/main/java/io/crate/session/RetryOnFailureResultReceiver.java

```diff
@@ -101,10 +96,6 @@ public void fail(Throwable wrappedError) {
         }
     }
 
-    private boolean indexWasTemporaryUnavailable(Throwable t) {
-        return t instanceof IndexNotFoundException inf && hasIndex.test(inf.getIndex().getName());
-    }
-
```

Comment:
Wasn't entirely sure about removing the `hasIndex` check, but the new `isShardNotAvailable` already includes `IndexNotFoundException` and we used `TransportActions.isShardNotAvailableException` also in a retry condition - it definitely can be a temporary error.

In case the table was deleted I suspect the retry would run into a `RelationUnknown` error anyway and abort the retries.